### PR TITLE
[WIP] git submodule for fzy-native

### PR DESCRIPTION
**Pull request by RishabhRD**
As @ruifm mentioned in #5, fzy-native sorter can be used as a submodule. It will
not only help in updating the modules easily, it will also give proper credits
to authors.

This pull request would also remove the fzy-native hard dependency.

Credits: @ruifm

### CODE REVIEW

**Review by ruifm — COMMENTED** [review]
Nice! Isn't this missing the actual entry for the git submodule in a `.gitmodules` file like:

```
[submodule "lua/popfix/fzy-native"]
	path = lua/popfix/fzy-native
	url = git@github.com:romgrk/fzy-lua-native.git
```
(not sure about the paths thought).

You can take a look at how tj does it in https://github.com/nvim-telescope/telescope-fzy-native.nvim

Also, which part of these changes makes fzy-lua-native optional?

**Comment by RishabhRD** [comment]
@ruifm Yeah! This was the initial commit just to make sure hard dependency of
fzy-lua-native is not there now.

I would just push another commit in a few minutes. That commit should have the
required files. Once done we can test it out.

**Comment by RishabhRD** [comment]
@ruifm I have added all necessary files that are needed to make this
work. I also added a script named install_fzy_native_sorter that is just a
wrapper around git commands and make command. It would make it easier to
install I guess.

If native sorter is not installed and anyone tries to use it, it prints a
warning message and fallbacks to default sorter.

I tried it for some scenarios and I feel it is complete enough.
What do you say? Do you feel is there something missing?

File: lua/popfix/sorter.lua [hunk 1]
```diff
@@ -36,6 +36,14 @@ function M:new_fzy_sorter(caseSensitive)
...
 
 function M:new_fzy_native_sorter(caseSensitive)
+	local fzy_native_file =
+	vim.api.nvim_get_runtime_file("external_modules/fzy_lua_native/lua/native.lua",
+	false)[1]
+	if fzy_native_file == nil then
+		print('fzy-lua-native sorter source is missing. You need to install fzy-lua-native source. Please refer README.md. Falling back to default sorter')
+		return nil
+	end
+	local fzy_native = loadfile(fzy_native_file)()
```
Comment (RishabhRD) [inline lua/popfix/sorter.lua:46]: This returns a function, not a table and hence important to call.

File: install_fzy_native_sorter [hunk 2]
```diff
@@ -0,0 +1,3 @@
+git submodule init
+git submodule update
+cd external_modules/fzy_lua_native/ && make
```
Comment (ruifm) [inline install_fzy_native_sorter:3]: IIRC most popular plugin managers already clone a repo with `--recurse-submodules` so this shouldn't be needed (unless installing manually).

And the make command could be done like:

```
make -C external_modules/fzy_lua_native
```

Comment (RishabhRD) [reply install_fzy_native_sorter]: I just noticed that, thanks! I updated the installation file and readme for the manual installation thing.

**Comment by RishabhRD** [comment]
@ruifm I feel we can merge it. The Makefile and readme are changed as your suggestion. Thanks for such great help. It would never be possible without your valuable suggestions.

This pull request needs your last approval to merge. :smiley_cat:

**Comment by RishabhRD** [comment]
@ruifm I don't know, you didn't reply for approval. I hope you are doing well
and not in any trouble.

I am merging the pull request as it was a major issue of stability.

If you feel something is missing please comment or make an issue.

Thanks for such a great help! :smiley:

**Pull request merged**